### PR TITLE
mgr/dashboard: build the dashboard from custom assets folder 

### PR DESCRIPTION
**Pull request by nizamial09**
Using this new Environment variable called
`DASHBOARD_BUILD_ASSETS_FROM`, we can chose from which folder under the dist/* to serve the dashboard from.

Inorder to properly use this, while building the dashboard we need to set DASHBOARD_BUILD_ASSETS_FROM=rhceph.

From the frontend, we need to manually build assets into a folder named rhceph under the dist/ folder. If the folder is present, then the dashboard frontend will be served from the rhceph folder.

Fixes: https://tracker.ceph.com/issues/58108
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by ktdreyer** [comment]
Do we set this environment variable at build time or at runtime? It's unclear from the description. The descriptions you've written use verbs like "loaded from", "build from" and "served from".

**Comment by nizamial09** [comment]
> Do we set this environment variable at build time or at runtime? It's unclear from the description. The descriptions you've written use verbs like "loaded from", "build from" and "served from".

This variable should be set at build time.

File: src/pybind/mgr/dashboard/controllers/home.py [hunk 1]
```diff
@@ -21,11 +21,12 @@
 
 class LanguageMixin(object):
     def __init__(self):
+        product_name = os.environ.get('DASHBOARD_BUILD_ASSETS_FROM', '')
         try:
             self.LANGUAGES = {
                 f
                 for f in os.listdir(mgr.get_frontend_path())
-                if os.path.isdir(os.path.join(mgr.get_frontend_path(), f))
+                if os.path.isdir(os.path.join(mgr.get_frontend_path(), product_name, f))
```
Comment (pereman2) [inline src/pybind/mgr/dashboard/controllers/home.py:29]: Why not using ceph options? I'm not familiar with rook but for example, you would have to set the env variable in every mgr host to let these mgr know where to find the dist folder? @nizamial09

Comment (pereman2) [reply src/pybind/mgr/dashboard/controllers/home.py:29]: Nevermind, it is a build process thing XD.

**Comment by ktdreyer** [comment]
> @ktdreyer This variable should be set at build time.

I am confused. I see the CMake change (for build time), but then the mgr also looks for this variable at runtime.

Which one is it?

**Pull request closed without merging**